istio-custom-ingressgateways/templates/autoscale.yaml

```
@@ -0,0 +1,19 @@
+{{- range $spec := .Values.global.customGateways }}
```

I would say the variable `spec` is overused and not very informative.
```suggestion
{{- range $customGateway := .Values.global.customGateways }}
```